### ISSUE TITLE
Canonical name doesn't match the scientific name in the API

### ISSUE THREAD

**Issue by ManonGros** [gbif/portal-feedback#4658]
This was an issue originally reported on helpdesk about this record: https://api.gbif.org/v1/species/157930679

```
  "scientificName": "Thalerastria diaphoraAH01",
  "canonicalName": "Thalerastria spec.",
```

The canonical name doesn't seem to be provided by the data provider: https://api.gbif.org/v1/species/157930679/verbatim, do we infer it on our side somehow?

Maybe a question for @thomasstjerne

**Comment by thomasstjerne** [comment]
Yes, I think this is because it is a [partially parsed name](https://api.gbif.org/v1/species/157930679/name) 
The parser understands that the name should be a species, but is not able to parse the epithet as this includes both capital letters and digits. It seems the parser then adds the "spec." in the canonical name meaning something like "a species in the genus Thalerastria"

**Comment by ManonGros** [comment]
Ok, thanks for checking!
Does it make sense? Shouldn't we have `Thalerastria diaphoraAH01` as canonical name? (If yes, then feel free to close the issue)

**Comment by thomasstjerne** [comment]
I think I´ll let @mdoering decide on that

**Comment by mdoering** [comment]
It is exacty like Thomas explained. The parser fails to understand the species eithet:
http://api.gbif.org/v1/parser/name?rank=species&name=Thalerastria%20diaphoraAH01

It could be anything there, an author, a remark some informal note. Should there maybe be a space between diaphora and AH01?

http://api.gbif.org/v1/parser/name?rank=species&name=Thalerastria%20diaphora%20AH01
